backend/src/gh.rs

```diff
@@ -445,4 +583,117 @@ impl GitHubClient {
+            bail!("Failed to fetch repository details: {}, Response: {}", status, response_text);
+        }
+    
+        // Deserialize the response to get the repository details
+        let repo_details: serde_json::Value = response.json().await?;
+    
+        // Retrieve the default branch from the response
+        let default_branch = repo_details["default_branch"]
+            .as_str()
+            .map(ToString::to_string)
+            .context("'default_branch' field missing in the response")?;
+    
+        Ok(default_branch)
+    }        
+
+    /// Fetches issues from the GitHub repository.
+    ///
+    /// This function retrieves issues from the specified repository using the GitHub API.
+    /// You can filter issues based on their state and associated labels.
+    ///
+    /// # Parameters:
+    /// - `state`: A string slice representing the state of the issues to fetch (e.g., "open", "closed", "all").
+    ///            Defaults to "open".
+    /// - `labels`: A comma-separated string slice representing labels to filter issues by. Defaults to `None`.
+    ///
+    /// # Returns:
+    /// A `Result<Vec<Value>>`:
+    /// - `Ok(issues)`: A vector of JSON values representing the issues fetched from the repository.
+    /// - `Err(e)`: An error message if the request fails or the response cannot be parsed.
+    ///
+    /// # Errors:
+    /// This function may return an error if:
+    /// - The `repo_url` is not in the expected format and cannot be parsed to derive the repository name.
+    /// - The request to fetch issues fails due to authentication issues, invalid input, or network problems.
+    /// - The GitHub API response cannot be parsed as a JSON array.
+    pub async fn get_issues(&self, state: Option<&str>, labels: Option<&str>) -> Result<Vec<Value>> {
+        let repo_name = self.get_repo_name().map_err(|e| {
+            error!("Failed to get repository name: {:?}", e);
+            e
+        })?;
+    
+        let state = state.unwrap_or("open"); // Default state
+        let mut query_params = vec![format!("state={}", state)];
+        if let Some(labels) = labels {
+            query_params.push(format!("labels={}", labels));
+        }
+        let query_string = format!("?{}", query_params.join("&"));
+    
+        let url = format!("{}/repos/{}/issues{}", GITHUB_API_URL, repo_name, query_string);
+        debug!("Request URL: {}", url);
+    
+        let response = self
+            .client
+            .get(&url)
+            .bearer_auth(&self.token)
+            .header("Accept", "application/vnd.github+json")
+            .header("User-Agent", "Hyde")
+            .timeout(std::time::Duration::from_secs(10))
+            .send()
+            .await?;
+    
+        if let Some(rate_limit_remaining) = response.headers().get("X-RateLimit-Remaining") {
+            debug!("GitHub API rate limit remaining: {}", rate_limit_remaining.to_str().unwrap_or("Unknown"));
+        }
```

Comment:
If we really are making API calls often enough for this to be an issue, what do you think about building an abstraction over the Github API that automatically does things like making sure we're staying under the ratelimit, and whatever other fancy shit?

This isn't required for this PR in any way, it would be an open issue for the future, this is spitballing.

Reply:
I did this for testing because I called it so much, I don't know what building an abstraction is, but I'm happy to learn and sounds fun.